src/components/StoreVersionDialog.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import { Component } from "react";
+import { Alert, Linking } from "react-native";
+import VersionCheck from "react-native-version-check";
+import { connect } from "react-redux";
+import type { Connector } from "react-redux";
+
+import { appNameIos, appIdIos } from "../constants/app";
+import { now, addDays, startOfDay, isBefore } from "../lib/date";
+import {
+  setAppUpdateAskLaterTime,
+  getAppUpdateAskLaterTime
+} from "../integrations/storage";
+import text from "../constants/text";
+import { notifyReporter } from "../actions/reporting";
+
+type VersionInfo = {
+  currentVersion: string,
+  isNeeded: boolean,
+  latestVersion: string
+};
+
+type Props = {
+  onError: Error => void
+};
+
+class StoreVersionDialog extends Component<Props> {
+  async componentDidMount() {
+    const versionInfo: VersionInfo = await VersionCheck.needUpdate();
+    const askLaterTime = await getAppUpdateAskLaterTime();
+    const updateMessageSuppressed =
+      !!askLaterTime && isBefore(now(), askLaterTime);
+
+    const showAppStoreAlert = versionInfo.isNeeded && !updateMessageSuppressed;
+    if (showAppStoreAlert) {
+      this.handleAppStoreAlert();
+    }
+  }
+
+  handleAppStoreAlert = () =>
+    Alert.alert(
```

Comment:
Will this work if the app was send to the backround while the update check was running?

Reply:
I did a quick test by throwing a delay into the code. It looks like when the app is in the background, nothing happens on the system level, but when you open the app again the dialog is there.

Reply:
Nice, that sounds good :+1: